### PR TITLE
8225571: Port DND source to use GTK instead of GDK

### PR DESCRIPTION
**Pull request by tsayao**
This is an alternative to #473. 

Why this is better?
1) less code - gtk handles it;
2) works on wayland;
3) gdk is constantly making API deprecated - gtk does not. Inside gdk version 3 there were a lot of breaking changes and deprecation - DND was made "managed", device is now "seat", etc..
4) feels more consistent with gnome desktop (it will use the same icons, cursor, etc);
5) probably will work with gtk4.

### CODE REVIEW

**Comment by tsayao** [comment]
Gtk2 working. Turns out it was just a wrong parameter (button number).

Not working yet: Drag view for gtk2.

**Comment by tsayao** [comment]
Drag view ok for gtk2 and gtk3.

**Comment by tsayao** [comment]
Now works on Ubuntu 16.04, 18.04 and 19.04 including wayland, with both gtk2 and gtk3.

Just a few little issues to fix.

**Comment by tsayao** [comment]
Current State:
- Works for all GTK versions
- Works with Wayland
- Works with Scenebuilder

Pending:
Some code do mouse/cursor grabs - this affects DND. Must investigate other scenarios.

**Comment by tsayao** [comment]
@kevinrushforth Should I open a jira issue for requesting review?

**Comment by tsayao** [comment]
@kevinrushforth Reduced the size o changes by reverting the unnecessary ones. The change only affects drag source now (and some necessary minor changes on other files). Functionality is unaffected.

**Comment by tsayao** [comment]
https://bugs.openjdk.java.net/browse/JDK-8225571

**Comment by tsayao** [comment]
To run tests:
./gradlew apps
java @build/run.args -cp apps/toys/DragDrop/dist/DragDrop.jar dragdrop.DragDropWithControls
java @build/run.args -cp apps/toys/DragDrop/dist/DragDrop.jar dragdrop.DragDropColor

**Comment by tsayao** [comment]
I consider this done until further requests. Did a lot of testing, all passed.

**Comment by tsayao** [comment]
@kevinrushforth I have significantly reduced the size of the patch and did lots of testing. I consider it ready for evaluation. If you agree, please remove the WIP tag.

**Comment by kevinrushforth** [comment]
@tsayao I'll take an initial look in the next couple of days.

**Comment by kevinrushforth** [comment]
I agree that this is in a state that we can remove "WIP". It might be worth doing a one-time sqaush/rebase/force push, since I don't know how much the previous history will help reviewers, and it will make a clean starting point for reviewing it? If so, this should be done before removing the WIP label (once people start reviewing the, it is best not to do a force push at all).

It is still unlikely we will have the bandwidth to review it in time to make openjfx13, but we can at least get the review started and see where we are.

**Comment by tsayao** [comment]
I can't find the whitespace Travis is complaining.

**Comment by tiainen** [comment]
The whitespace issues are caused by spaces on empty lines. For example, this commit introduced a few of those: https://github.com/tsayao/openjdk-jfx/commit/fd93b6f074f902ec9d82b0dc5ed27a232ee29a86#diff-3e4475de661c80010beb8b01998ae9a8R46

Removing the whitespace on otherwise empty lines should fix the issue.

**Comment by kevinrushforth** [comment]
Hopefully that fixes it. If you want an easy way of doing it in the future:

1. Generate a list of files that you want to check and fix (either manually or using `git diff --name-only develop` if your branch is based on and up-to-date with `develop`) and put it in `mod-files.txt`.

2. Run the following command:

```
    cat mod-files.txt | bash tools/scripts/checkWhiteSpace -F -x -S
```

File: modules/javafx.graphics/src/main/native-glass/gtk/glass_dnd.cpp [hunk 1]
```diff
@@ -141,6 +141,12 @@ static struct {
 
 gboolean is_dnd_owner = FALSE;
 
+GtkWidget *drag_widget = NULL;
```
Comment (tsayao) [inline modules/javafx.graphics/src/main/native-glass/gtk/glass_dnd.cpp]: A GtkWindow is used offscreen to handle the signals gtk sends.

**Comment by kevinrushforth** [comment]
This looks ready to review, so removing the [WIP] label. Given other priorities, it will very likely not make openjfx13, but we'll put it on list to review.

@pankaj-bansal can you review it as well when you get time?

**Comment by tsayao** [comment]
Note that its an Enhacement but fixes:

- https://bugs.openjdk.java.net/browse/JDK-8227004
- Drag and Drop on Wayland (wayland is not ready for production yet, so not very important - but seems to be the future).

**Comment by kevinrushforth** [comment]
@tsayao Thanks for the update. I'll make sure this is on my review queue for the near future, but it won't be until after the Skara migration.

**Comment by kevinrushforth** [comment]
As announced in [this message](https://mail.openjdk.java.net/pipermail/openjfx-dev/2019-September/023551.html), the official `jfx` repository is moving from [hg.openjdk.java.net/openjfx/jfx-dev/rt](https://hg.openjdk.java.net/openjfx/jfx-dev/rt) to [github.com/openjdk/jfx](https://github.com/openjdk/jfx).

This [sandbox repository](https://github.com/javafxports/openjdk-jfx) is being retired on 1-Oct-2019. You will need to migrate your pull request to the openjdk/jfx repo in order for this review to continue. Here are [instructions for migrating your pull request](https://mail.openjdk.java.net/pipermail/openjfx-dev/2019-September/023560.html). The updated [CONTRIBUTING.md](https://github.com/openjdk/jfx/blob/master/CONTRIBUTING.md) doc has additional information on how to proceed.

Once you have done this, it would be helpful to add a comment with a pointer to the new PR.

**Comment by kevinrushforth** [comment]
Moved to: [openjdk/jfx/pull/4](https://github.com/openjdk/jfx/pull/4)

**Pull request closed without merging**